diff -rupN a/src/mapreduce.cpp b/src/mapreduce.cpp
--- a/src/mapreduce.cpp 2011-06-20 12:09:30.000000000 -0400
+++ b/src/mapreduce.cpp 2013-11-11 16:23:15.984789767 -0500
@@ -2697,7 +2697,7 @@ void MapReduce::findfiles(char *str, int
   struct stat buf;
   char newstr[MAXLINE];
 
-  err = stat(str,&buf);
+  err = lstat(str,&buf);
   if (err) {
	 char msg[256];
	 sprintf(msg,"Could not query status of file %s in map",str);
@@ -2721,6 +2721,7 @@ void MapReduce::findfiles(char *str, int
		findfiles(newstr,recurse,readflag,nfile,maxfile,files);
	 }
	 closedir(dp);
+  } else if (S_ISLNK(buf.st_mode)) {
   } else {
	 char msg[256];
	 sprintf(msg,"Invalid filename %s in map",str);
