diff -rupN a/src/cmapreduce.h b/src/cmapreduce.h
--- a/src/cmapreduce.h	2011-01-27 17:11:45.000000000 -0500
+++ b/src/cmapreduce.h	2013-10-22 22:47:06.829436153 -0400
@@ -48,12 +48,12 @@ uint64_t MR_map(void *MRptr, int nmap,
 uint64_t MR_map_add(void *MRptr, int nmap,
 		    void (*mymap)(int, void *KVptr, void *APPptr),
 		    void *APPptr, int addflag);
-uint64_t MR_map_file(void *MRptr, int nstr, char *strings,
+uint64_t MR_map_file(void *MRptr, int nstr, char **strings,
 		     int self, int recurse, int readfile,
 		     void (*mymap)(int, char *, 
 				   void *KVptr, void *APPptr),
 		     void *APPptr);
-uint64_t MR_map_file_add(void *MRptr, int nstr, char *strings,
+uint64_t MR_map_file_add(void *MRptr, int nstr, char **strings,
 			 int self, int recurse, int readfile,
 			 void (*mymap)(int, char *, 
 				       void *KVptr, void *APPptr),
